app/src/androidTest/java/com/android/sample/screen/MainPageTest.kt

```diff
@@ -0,0 +1,85 @@
+package com.android.sample.screen
+
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTouchInput
+import androidx.compose.ui.test.swipeLeft
+import androidx.compose.ui.unit.dp
+import androidx.test.espresso.intent.Intents
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import com.android.sample.ui.swipePage.ReceipeDisplay
+import com.android.sample.ui.swipePage.SwipePage
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+
+@RunWith(AndroidJUnit4::class)
+class MainPageTest : TestCase() {
+  private lateinit var navigationActions: NavigationActions
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    navigationActions = mock(NavigationActions::class.java)
+    `when`(navigationActions.currentRoute()).thenReturn(Route.MAIN)
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  fun swipePageCorrectlyDisplayed() {
+    composeTestRule.setContent {
+      SwipePage(navigationActions) // Set up the SignInScreen directly
+    }
+    composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("topBar").assertIsDisplayed()
+  }
+
+  @Test
+  fun recipeAndDescriptionAreCorrectlyDisplayed() {
+    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }
+    composeTestRule.onNodeWithTag("recipeDescription").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("recipeImage").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("swipeUIDescription").assertIsDisplayed()
+  }
+
+  @Test
+  fun testBiggerDescription() {
+    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }
+
+    // Make sure the screen is displayed
+    composeTestRule.onNodeWithTag("displayDescription").assertIsDisplayed()
+
+    // Simulate a drag event
+    composeTestRule.onNodeWithTag("displayDescription").performClick()
+    composeTestRule.onNodeWithTag("RecipeEntireDescription").assertIsDisplayed()
+  }
+
+  @Test
+  fun testDraggingEvent() {
+    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }
+
+    // Make sure the screen is displayed
+    composeTestRule.onNodeWithTag("draggableItem").assertIsDisplayed()
+
+    // Simulate a drag event
+    composeTestRule.onNodeWithTag("draggableItem").performTouchInput {
+      swipeLeft(durationMillis = 5000)
+    }
```

Comment:
### Review 4: Toast Messages

**Review Comment:**
There are no tests covering the Like/Dislike Toast feedback messages that appear when swiping. Consider adding tests to ensure the correct feedback is displayed to users after interacting with the swipe gestures.

**Code Snippet:**
```kotlin
@Test
fun testLikeToastMessage() {
    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }

    // Simulate a left swipe event
    composeTestRule.onNodeWithTag("draggableItem").performTouchInput {
        swipeLeft()
    }

    // Check if the "Like" toast message is displayed
    composeTestRule.onNodeWithText("Like").assertIsDisplayed()
}

@Test
fun testDislikeToastMessage() {
    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }

    // Simulate a right swipe event
    composeTestRule.onNodeWithTag("draggableItem").performTouchInput {
        swipeRight()
    }

    // Check if the "Dislike" toast message is displayed
    composeTestRule.onNodeWithText("Dislike").assertIsDisplayed()
}
```

Reply:
I won't be testing the toast since I plan to modify this part and replace it with the ViewModel code, which I will test directly after this PR is merged. The toast was only used as a temporary measure to check if it worked while testing in the emulator.

Reply:
Thank you for the update. I understand that the toast was a temporary measure and that you plan to replace it with ViewModel code. We will wait for the update with the ViewModel to perform the necessary tests.